### ISSUE TITLE
It's clear now that e-mail from Facebook is verified

### ISSUE THREAD

**Issue by DominikSerafin** [pennersr/django-allauth#1126]
I don't know django-allauth so well to make the pull request myself but someone should probably change 'VERIFIED_EMAIL' option of Facebook settings and documentation on this.

https://stackoverflow.com/questions/14280535/is-it-possible-to-check-if-an-email-is-confirmed-on-facebook

Right now doc says this:
>
VERIFIED_EMAIL:
It is not clear from the Facebook documentation whether or not the fact that the account is verified implies that the e-mail address is verified as well. For example, verification could also be done by phone or credit card. To be on the safe side, the default is to treat e-mail addresses from Facebook as unverified. But, if you feel that is too paranoid, then use this setting to mark them as verified. Due to lack of an official statement from the side of Facebook, attemps have been made to reverse engineer the meaning of the verified flag. Do know that by setting this to True you may be introducing a security risk.

**Comment by bittner** [comment]
So, what should that text say [in the docs](http://django-allauth.readthedocs.org/en/latest/providers.html#facebook)?

**Comment by lukeburden** [comment]
Today I ran a quick experiment:

1) I created a functional Facebook account using a cell number and no email address. During authentication (using an app using allauth) no email was passed back.
2) I then added an email address to the Facebook account from 1), but did *not* verify it (ie, I did not click on the link emailed to me by Facebook). During authentication I again received no email data.
3) Finally, I confirmed the email on Facebook's side by clicking on the supplied link emailed to me. During authentication, I finally received the email address as part of the Facebook user's data.

If a hacker can't create a new Facebook account with an unverified email being exposed via the graph API, then the vector for that attack has effectively been mitigated.

Now, Facebook are not communicating well the reality on their end so for some applications I'd still feel uneasy trusting Facebook's email verification. That's your call based on the security needs and specific use of your application. But for most applications it would seem fine to me to trust Facebook emails as verified at this stage.

@pennersr, thoughts on which way allauth should go with this in terms of the default behaviour?
